### PR TITLE
CXF-SOAP: Cover possible regression prior CXF fix causing indefinitive hang

### PR DESCRIPTION
**Pull request by JiriOndrusek**
fixes https://github.com/apache/camel-quarkus/issues/5682

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: integration-test-groups/cxf-soap/cxf-soap-ws-security-client/src/test/java/org/apache/camel/quarkus/component/cxf/soap/wss/client/it/CxfSoapWssClientTest.java [hunk 1]
```diff
@@ -86,6 +99,57 @@ void wsdlUpToDate() throws IOException {
...
+        final Service service = Service.create(serviceUrl, SayHelloService.SERVICE);
+
+        SayHelloWrongWS port = service.getPort(SayHelloWrongWS.class);
+        BindingProvider bp = (BindingProvider) port;
+
+        Map<String, Object> requestContext = bp.getRequestContext();
+
+        //non-existing server url is used
+        requestContext.put(BindingProvider.ENDPOINT_ADDRESS_PROPERTY,
+                String.format("http://localhost:%d/soapservice/sayHelloWrong", AvailablePortFinder.getNextAvailable()));
```
Comment (jamesnetherton) [inline integration-test-groups/cxf-soap/cxf-soap-ws-security-client/src/test/java/org/apache/camel/quarkus/component/cxf/soap/wss/client/it/CxfSoapWssClientTest.java:139]: Nitpick - if we don't actually care about the server URL, can we just hard code the port to any value? If so we can remove the dependency on `camel-test-junit5`.

**Review by jamesnetherton — APPROVED** [review]

**Comment by jamesnetherton** [comment]
@JiriOndrusek do you mind rebasing on latest `main` just to check it with the latest upgrades.

I left an optional nitpick comment too.

File: integration-test-groups/cxf-soap/cxf-soap-ws-security-client/src/test/java/org/apache/camel/quarkus/component/cxf/soap/wss/client/it/CxfSoapWssClientTest.java (hunk 1 above)
Comment (JiriOndrusek) [reply integration-test-groups/cxf-soap/cxf-soap-ws-security-client/src/test/java/org/apache/camel/quarkus/component/cxf/soap/wss/client/it/CxfSoapWssClientTest.java:139]: I understand your point. I used the next available port just to be sure that the port is really not used. Maybe it is not necessary. I'll remove the dependency and hardcode a fixed value

**Comment by JiriOndrusek** [comment]
PR is rebased the dependency `camel-test-junit5` removed

**Review by zhfeng — APPROVED** [review]
Nice work!

**Pull request merged**